### PR TITLE
Percentile_rank and related others

### LINKED ISSUE #190: Add MicroSeries.{percentile, decile, quintile, quartile}_rank
Based off of recently-added `MicroSeries.rank(pct=True)`

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
Fixes #190

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Could you add tests?

File: microdf/generic.py [hunk 1]
```diff
@@ -645,7 +663,7 @@ def groupby(self, by: Union[str, list], *args, **kwargs):
         """
         self["__tmp_weights"] = self.weights
         gb = super().groupby(by, *args, **kwargs)
-        weights = gb["__tmp_weights"]
+        weights = copy.deepcopy(gb["__tmp_weights"])
```
Comment (MaxGhenis) [inline microdf/generic.py:666]: Would `gb["__tmp_weights"].copy()` work here? If it's a `pd.Series` then these are equivalent, and could avoid the extra import.

**Review by MaxGhenis — COMMENTED** [review]
One small thing otherwise I can merge

File: microdf/generic.py (hunk 1 above)
Comment (nikhilwoodruff) [reply microdf/generic.py:666]: Seems like it won't, just tried. ```SeriesGroupBy``` doesn't have a ```copy``` method.

**Pull request merged**